src/c2goto/headers/ubuntu20.04/kernel_5.15.0-76/include/asm/uaccess.h

```diff
@@ -0,0 +1,14 @@
+
+#define PAGE_SIZE 4096 // page size
+#define USER_MEMORY_SPACE 10000 //only for simulate the user memory
+#define KERNEL_MEMORY_SPACE 10000// only for simulate the kernel memory
+
+char user_memory[USER_MEMORY_SPACE]; //mock user memory
+char kernel_memory[KERNEL_MEMORY_SPACE];//mock user memory
+
+//simulate copy_to_user function in kernel space 
+unsigned long copy_to_user(void* to, void* from, unsigned size);
+unsigned long copy_from_user(void* to, void* from, unsigned size);
```

Comment:
According to my (Linux) sources, both of these functions take their `size` parameter as `unsigned long`.

Reply:
Also, the 2nd and 3rd question, i put those header file because in linux source codes, those are the dependency headers for uacess.h, although my headers did not use function or variables from those files. I did not put them in CmakeLIist since i feel bit weird to put those empty files in the bundle. So maybe I should remove them, or put them in Cmakelist? Yes for the 3rd question,so maybe I can create a new file in c2goto library directory and put all kernel functions in ?

Reply:
> I did not put them in CmakeLIist since i feel bit weird to put those empty files in the bundle. So maybe I should remove them, or put them in Cmakelist?

There is nothing wrong with bundling empty files, if that's how they are to be used. For headers it might make sense to do that in order to hide system headers with the same include path. But having empty files in ESBMC's source tree that are not used indeed feelds weird.

Reply:
> Yes for the 3rd question,so maybe I can create a new file in c2goto library directory and put all kernel functions in ?

Yes, that would be great. Just put it in the `c2goto/library` directory for now.